### PR TITLE
Added an Event logger panel

### PR DESCRIPTION
**Pull request by javier-tarazaga**
### Description of the Change

- Added a new `Event Log` panel. In order to put messages into the panel, you simply need to call the eventLog.actions/addEventLogRow with a LogLevel and a message and will render in the panel. 
- Included a button to delete all the current items logged in the panel
- Added a bunch of styling fixes related to the panels and a few style small modifications to them. 
- Removed icons from the panel header. 

### Benefits

Now we have a panel in which we can display messages to the user related to things the IDE is doing or whatever we consider relevant adding to it (per example when/if introducing plugins) 

### Verification Process

- Run the applications and open the event logger panel at the bottom right corner of the IDE.

### CODE REVIEW

File: src/components/projecteditor/ProjectEditor.tsx [hunk 1]
```diff
@@ -118,7 +119,7 @@ export class ProjectEditor extends React.Component<IProps, IState> {
                                         <Panes dragging={sidePanelDragging} />
 
                                         { this.isPanelOpen(Panels.Transactions) &&
-                                        <Panel icon={ <IconTransactions /> } name='Transactions History' onClose={() => closePanel(Panels.Transactions)} dragging={sidePanelDragging}>
+                                        <Panel name='Transactions History' onClose={() => closePanel(Panels.Transactions)} dragging={sidePanelDragging}>
```
Comment (ostap1010) [inline src/components/projecteditor/ProjectEditor.tsx]: Question here - is it as intended to remove icons from panels?

File: src/components/projecteditor/style.less [hunk 2]
```diff
@@ -141,15 +133,18 @@
     }
 
     .sideButton {
-        border-color: transparent;
         padding: 5px;
         transition: all .15s ease;
-        border-radius: 0px;
```
Comment (Spejbl) [inline src/components/projecteditor/style.less]: Is this one intended? Buttons in panels now have border-radius again after removing this style

File: src/components/projecteditor/panels/explorer/items/style.less [hunk 3]
```diff
@@ -26,7 +26,7 @@ select {
     user-select: none;
     display: block;
     color: #fff;
-    line-height: 28px;
```
Comment (Spejbl) [inline src/components/projecteditor/panels/explorer/items/style.less]: Explorer feels too much compact on non-retina display. I would consider to have line-height at least 24-25px.

**Comment by filippsen** [comment]
@Spejbl the previous deployer version contained a runtime check-and-swap action that would choose between the internal _web3_ object and the externally provided one. 

Now, when the account is set to `external` (or `'metamask'` ?), the new deployer is probably assuming `window.web3` is always valid. 
See also:
> ./src/services/deployer/deployRunner.ts:37:        this.currWeb3 = account.type === 'metamask' ? window.web3 : getWeb3(this.environment.endpoint);

File: src/components/projecteditor/ProjectEditor.tsx (hunk 1 above)
Comment (javier-tarazaga) [reply src/components/projecteditor/ProjectEditor.tsx]: Yeap, it is intended.

File: src/components/projecteditor/style.less (hunk 2 above)
Comment (javier-tarazaga) [reply src/components/projecteditor/style.less]: Actually not! Let me remove that one

File: src/components/projecteditor/panels/explorer/items/style.less (hunk 3 above)
Comment (javier-tarazaga) [reply src/components/projecteditor/panels/explorer/items/style.less]: Sure. In fact, I thought it did have 24px but let me force it instead through this prop.

**Comment by javier-tarazaga** [comment]
Thanks @Spejbl for raising that flag but as you said, in fact, I haven't changed that in this PR. Let's add a new issue to the TODO list to make sure this is covered (this will be impacted by @ostap1010 work anyway), but we can set it up as self-reminder to make sure it works as expected.

File: src/components/projecteditor/ProjectEditor.tsx (hunk 1 above)
Comment (javier-tarazaga) [reply src/components/projecteditor/ProjectEditor.tsx]: Added it to the PR description

File: src/components/projecteditor/style.less (hunk 2 above)
Comment (javier-tarazaga) [reply src/components/projecteditor/style.less]: I will remove it from the btnNoBg global style instead.

**Pull request merged**